### ISSUE TITLE
[NumberInput] Support of different decimal separators

### ISSUE THREAD

**Issue by ruessej** [mantinedev/mantine#521]
- Usage example, including component/hook API: use , instead of a . as decimal separator
- Possible implementation – describe how feature can be implemented: You could specify an object with the separator or an enum where you have fixed formats
- Do you want to contribute this feature and create a pull request? I will try it

**Comment by rtivital** [comment]
What prop will it be to override separator, I do not understand how we can use an object, all I can think of is a string:
```tsx
<NumberInput decimalSeparator="," />
```

**Comment by ruessej** [comment]
Yes you right i though of the thousands separator too

```ts
<NumberInput separators={ decimal: ",", thousands: "." } />
```

but when i think now about it we dont need that separator or?

**Comment by rtivital** [comment]
It seems like a hard thing to do, but feel free to try with your proposed api

**Comment by jerebtw** [comment]
@rtivital is it a problem when all . are replaced with the custom separator?

**Comment by LucasLemanowicz** [comment]
I suggest looking into the built-in JS `Intl.NumberFormat` API: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat

Would provide good extendability beyond just that one specific use case and let you do more with less code.

**Comment by ruessej** [comment]
Okay i will look in to it later

**Comment by ruessej** [comment]
The simplest solution would be to let the user create their own parse and format function.
But I don't know if this is too much for this simple API.

**Comment by rtivital** [comment]
Fixed in 3.5.0
